### PR TITLE
Upgrade to sailor 2.6.18

### PR DESCRIPTION
**Pull request by denyshld**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.0.2 (October 30, 2020)
+
+* Upgrade to sailor 2.6.18
```
Comment (shulkaolka) [inline CHANGELOG.md:3]:
```suggestion
* Upgrade to sailor 2.6.18
* Annual audit of the component code to check if it exposes a sensitive data in the logs
```

File: circle.yml [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
-    version: 5.0.0
-    version: 6.0.0
-test:
-  override:
-    - npm test
+version: 2
+jobs:
+  test:
+    docker:
+      - image: circleci/node:12-stretch
```
Comment (shulkaolka) [inline circle.yml:5]:
```suggestion
      - image: circleci/node:14-stretch
```

File: lib/actions/unsubscribe.js [hunk 3]
```diff
@@ -26,13 +26,13 @@ function processAction(msg, conf) {
         mailchimp.setApiKey(apiKey);
         const hash = md5(body.email.toLowerCase());
         const resource = `/lists/${listId}/members/${hash}`
-        self.logger.info('Trying to delete the resource %s',resource);
+        self.logger.info('Trying to delete the resource');
         const response = yield mailchimp.delete('/lists/' + listId + '/members/' + hash);
-        self.logger.info('Received response response=%j', response);
+        self.logger.info('Received response');
```
Comment (shulkaolka) [inline lib/actions/unsubscribe.js:31]:
```suggestion
        self.logger.info('Response received');
```

File: lib/actions/unsubscribe.js [hunk 4]
```diff
@@ -26,13 +26,13 @@ function processAction(msg, conf) {
...
         this.emit('data', messages.newMessageWithBody(response));
         this.emit('end');
     }.bind(this)).catch(err => {
-        self.logger.error('Error occured', err.stack || err);
+        self.logger.error('Error occured');
```
Comment (shulkaolka) [inline lib/actions/unsubscribe.js:35]:
```suggestion
        self.logger.error('Error occurred');
```

File: lib/actions/upsertCustomer.js [hunk 5]
```diff
@@ -33,14 +33,14 @@ function processAction(msg, conf) {
         mailchimp.setApiKey(apiKey);
         // Check the opt_in_status in body, it is string but should be boolean
         body.opt_in_status = body.opt_in_status === "true";
-        self.logger.info('Pushing data to mailchimp payload=%j', body);
+        self.logger.info('Pushing data to mailchimp payload...');
         const response = yield mailchimp.put('/ecommerce/stores/' + storeId + '/customers/' + body.id, body);
-        self.logger.info('Received response response=%j', response);
+        self.logger.info('Received response');
```
Comment (shulkaolka) [inline lib/actions/upsertCustomer.js:38]:
```suggestion
        self.logger.info('Response received');
```

File: lib/actions/upsertCustomer.js [hunk 6]
```diff
@@ -33,14 +33,14 @@ function processAction(msg, conf) {
...
         this.emit('data', messages.newMessageWithBody(response));
         this.emit('end');
     }.bind(this)).catch(err => {
-        self.logger.error('Error occured', err.stack || err);
+        self.logger.error('Error occured');
```
Comment (shulkaolka) [inline lib/actions/upsertCustomer.js:42]:
```suggestion
        self.logger.error('Error occurred');
```

File: package.json [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
   "name": "mailchimp",
-  "version": "1.0.1",
+  "version": "1.0.2",
   "description": "elastic.io component that connects to Mailchimp API (node.js)",
   "scripts": {
     "test": "./node_modules/.bin/grunt test"
   },
   "engines": {
-    "node": ">=10"
+    "node": ">=12.13.0"
```
Comment (shulkaolka) [inline package.json:9]: Maybe v14 will be better?

**Review by shulkaolka — CHANGES_REQUESTED** [review]

**Review by shulkaolka — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,8 @@
+## 1.0.2 (November 5, 2020)
+
+* Upgrade to sailor 2.6.18
+* Annual audit of the component code to check if it exposes a sensitive data in the logs
+
```
Comment (shulkaolka) [inline CHANGELOG.md:5]:
```suggestion
* Annual npm vulnerabilities audit

```

File: lib/actions/subscribe.js [hunk 9]
```diff
@@ -1,64 +1,65 @@
...
-        this.emit('error', err);
-        this.emit('end');
-    });
-}
\ No newline at end of file
+    self.logger.info('Pushing data to mailchimp payload...');
+    const hash = md5(body.email.toLowerCase());
+    const response = await mailchimp.put(`/lists/${listId}/members/${hash}`, payload);
+    self.logger.info('Received response');
+    this.emit('data', messages.newMessageWithBody(response));
```
Comment (shulkaolka) [inline lib/actions/subscribe.js:55]:
```suggestion
    await this.emit('data', messages.newMessageWithBody(response));
```

File: lib/actions/unsubscribe.js [hunk 10]
```diff
@@ -1,39 +1,40 @@
...
+    if (!body.email) {
+      throw new Error('Error in incoming data, required parameter email is missing');
+    }
+    mailchimp.setApiKey(apiKey);
+    const hash = md5(body.email.toLowerCase());
+    const resource = `/lists/${listId}/members/${hash}`;
+    self.logger.info('Trying to delete the resource');
+    const response = await mailchimp.delete(resource);
+    self.logger.info('Response received');
+    this.emit('data', messages.newMessageWithBody(response));
```
Comment (shulkaolka) [inline lib/actions/unsubscribe.js:30]:
```suggestion
    await this.emit('data', messages.newMessageWithBody(response));
```

File: lib/actions/upsertCustomer.js [hunk 11]
```diff
@@ -1,46 +1,45 @@
...
+    if (!conf.apiKey) {
+      throw Error('Error in incoming configuration, required parameter apiKey is missing');
+    }
+    mailchimp.setApiKey(apiKey);
+    // Check the opt_in_status in body, it is string but should be boolean
+    body.opt_in_status = body.opt_in_status === 'true';
+    self.logger.info('Pushing data to mailchimp payload...');
+    const response = await mailchimp.put(`/ecommerce/stores/${storeId}/customers/${body.id}`, body);
+    self.logger.info('Response received');
+    this.emit('data', messages.newMessageWithBody(response));
```
Comment (shulkaolka) [inline lib/actions/upsertCustomer.js:36]:
```suggestion
    await this.emit('data', messages.newMessageWithBody(response));
```

File: package.json [hunk 12]
```diff
@@ -1,12 +1,13 @@
...
+  "version": "1.0.2",
   "description": "elastic.io component that connects to Mailchimp API (node.js)",
   "scripts": {
-    "test": "./node_modules/.bin/grunt test"
+    "pretest": "eslint verifyCredentials.js lib spec --fix",
+    "test": "mocha spec/** --timeout 50000"
   },
   "engines": {
-    "node": ">=10"
+    "node": ">=14"
```
Comment (shulkaolka) [inline package.json:10]:
```suggestion
    "node": "14"
```

**Review by shulkaolka — CHANGES_REQUESTED** [review]

**Review by shulkaolka — APPROVED** [review]

**Review by A3a3e1 — APPROVED** [review]

**Pull request merged**